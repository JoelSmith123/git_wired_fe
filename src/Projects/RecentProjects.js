
import React, { Component } from 'react';
import ProjectCard from './ProjectCard.js'

import UserSession from '../tools/UserSession.js'


export default class RecentProjects extends Component {

  constructor = () => {
    this.user = new UserSession
  }

  // TO DO - TEST ME
  getProjects() {
    // TO DO - Confirm ENDPOINT
    let url = 'https://git-wired-be.herokuapp.com/api/v1/projects'
    fetch(url, {
        method:  "GET",
        headers: { "Content-Type": "application/json", },
        body:    JSON.stringify(this.user.getGitWiredToken()),
    })
    .then(response => response.json() )
    .then(data     => this.renderProjectCards(data).bind(this))
    .catch(error   => console.error('Error:', error));
  }

  projects(data) {
    return data['projects']
  }

  // TO DO - TEST ME
  renderProjectCards(data){
    let projects = this.projects(data)
    let l = projects.length
    for(let i = 0; i < l; i++) {
      let card = projects[i]
      return <ProjectCard project={card} />
    }
  }

  // TO DO - TEST ME
  render() {
    return (
      <div className='RecentProjects'>
        { this.renderProjectCards() }
      </div>
    )
  }




}
