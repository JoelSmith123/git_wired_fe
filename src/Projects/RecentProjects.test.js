import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import RecentProjects from './RecentProjects.js'

describe('RecentProjects', () => {
  let data
  let mockProjects
  beforeEach(() => {
    mockProjects = [
      { 'projectTitle': 'Title1',
        'repoName':     'RepoName1',
        'createdAt':    'Created1',
        'updatedAt':    'Updated1',
      },
      { 'projectTitle': 'Title2',
        'repoName':     'RepoName2',
        'createdAt':    'Created2',
        'updatedAt':    'Updated2',
      },
      { 'projectTitle': 'Title3',
        'repoName':     'RepoName3',
        'createdAt':    'Created3',
        'updatedAt':    'Updated3',
      },
    ]

    data = {
      'id': '123',
      'projects': mockProjects,
    }
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RecentProjects />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<RecentProjects />)
    expect(wrapper).toMatchSnapshot()
  }) 

  xit('return projects', () => {
    // let projects = recents.projects(data)
    // let card     = projects[0]
    // expect(projects).to.be.an('array')
    // expect(card).to.be.an('object')
    // expect(card.projectTitle).to.equal('Title1')
    // expect(card.repoName    ).to.equal('RepoName1')
    // expect(card.createdAt   ).to.equal('Created1')
    // expect(card.updatedAt   ).to.equal('Updated1')

  });

  it.skip('render Project Cards', done => {

    done();
  });

  it.skip('renders', done => {

    done();
  });


});
