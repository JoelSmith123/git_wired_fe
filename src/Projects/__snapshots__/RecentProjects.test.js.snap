// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentProjects should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecentProjects />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ProjectCard
          project={
            Object {
              "createdAt": "Created At",
              "projectTitle": "Wireframe Title 1",
              "repoName": "Repo 1",
              "updatedAt": "Updated At",
            }
          }
        />,
        <ProjectCard
          project={
            Object {
              "createdAt": "Created At",
              "projectTitle": "Wireframe Title 2",
              "repoName": "Repo 2",
              "updatedAt": "Updated At",
            }
          }
        />,
      ],
      "className": "RecentProjects",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "project": Object {
            "createdAt": "Created At",
            "projectTitle": "Wireframe Title 1",
            "repoName": "Repo 1",
            "updatedAt": "Updated At",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "project": Object {
            "createdAt": "Created At",
            "projectTitle": "Wireframe Title 2",
            "repoName": "Repo 2",
            "updatedAt": "Updated At",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ProjectCard
            project={
              Object {
                "createdAt": "Created At",
                "projectTitle": "Wireframe Title 1",
                "repoName": "Repo 1",
                "updatedAt": "Updated At",
              }
            }
          />,
          <ProjectCard
            project={
              Object {
                "createdAt": "Created At",
                "projectTitle": "Wireframe Title 2",
                "repoName": "Repo 2",
                "updatedAt": "Updated At",
              }
            }
          />,
        ],
        "className": "RecentProjects",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "project": Object {
              "createdAt": "Created At",
              "projectTitle": "Wireframe Title 1",
              "repoName": "Repo 1",
              "updatedAt": "Updated At",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "project": Object {
              "createdAt": "Created At",
              "projectTitle": "Wireframe Title 2",
              "repoName": "Repo 2",
              "updatedAt": "Updated At",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
